### PR TITLE
feat: remove now redundant parts of the HTTP API

### PR DESCRIPTION
**Pull request by tustvold**
I've left the parts of the HTTP API that don't yet have gRPC equivalents

### CODE REVIEW

**Review by pauldix — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -192,24 +192,12 @@ Each IOx instance requires a writer ID.
...
-- send an HTTP PUT request:
-```
-curl --request PUT \
-  --url http://localhost:8080/iox/api/v1/id \
-  --header 'Content-Type: application/json' \
-  --data '{
-  "id": 42
-  }'
-```
+- set using the CLI: `influxdb_iox writer set 42` or `cargo run -- writer set 42`
```
Comment (alamb) [inline README.md:195]: 👍

File: README.md [hunk 2]
```diff
@@ -229,7 +217,7 @@ To query stored data, use the `/api/v2/read` endpoint with a SQL query. This exa
 all data in the `company` organization's `sensors` bucket for the `processes` measurement:
 
 ```shell
-curl -v -G -d 'org=company' -d 'bucket=sensors' --data-urlencode 'sql_query=select * from processes' "http://127.0.0.1:8080/api/v2/read"
+curl -v -G --data-urlencode 'q=select * from processes' -d 'format=pretty' "http://127.0.0.1:8080/iox/api/v1/databases/company_sensors/query"
```
Comment (alamb) [inline README.md:220]: thank you for fixing that

**Review by alamb — APPROVED** [review]
Nice work @tustvold

**Review by domodwyer — APPROVED** [review]
Can we not get rid of `ApiErrorCode` yet? 🔥

**Pull request merged**